ipa-core/src/protocol/context/mod.rs

```diff
@@ -333,38 +362,170 @@ impl<'a> Inner<'a> {
     }
 }
 
+/// Reshards the given set of shares across all shards available on the current helper. It uses the
+/// provided function to determine destination shards for each input row.
+///
+/// The resulting vector contains all shares sent to this shard by other shards and shares selected
+/// locally by `shard_picker` fn to stay on this shard.
+///
+/// Resharding opens one communication channel per peer (N^2 channels will be open in total,
+/// N per shard). Each channel stays open until the very last row is processed, then they are explicitly
+/// closed, even if nothing has been communicated between that pair.
+///
+/// ## Panics
+/// It does not panic
```

Comment:
It would panic if `shard_picker` returned an out-of-range shard index, which seems like it is possible given that `ShardIndex` allows construction from arbitrary integers.

(This is admittedly a nitpick -- perhaps a more salient question is whether the clippy lints for error / panic docs are worthwhile.)

Reply:
It is annoying sometimes but it could be useful to see all places where things may panic, often w/o intention to have it